pkg/osquery/extension.go

```diff
@@ -563,6 +563,29 @@ func (e *Extension) enrolled() bool {
 	return nodeKey != ""
 }
 
+// nodeKey returns the current node key for this registration. It ensures that if the
+// node key is set in the database but not in the extension, we will pull it. We should therefore
+// always use this function when we need the node key to make requests to the device server,
+// to avoid making requests with an empty node key and attempting to reenroll instead.
+func (e *Extension) nodeKey() string {
```

Comment:
Do we need it in both places? Can we move it out of the extension?

Reply:
If we pulled it out of the extension struct, then we'd have to be making a db call to fetch the node key for every single call to the device server -- I assumed we wouldn't want this, but maybe it's no big deal

Reply:
I don't believe it should affect other parts of the enroll flow, but maybe I'm missing something. I'll put in a PR to adjust naming/comments

Reply:
https://github.com/kolide/launcher/pull/2474 lmk!